code/modules/admin/verbs/secrets.dm

```diff
@@ -348,16 +348,40 @@ GLOBAL_DATUM(everyone_a_traitor, /datum/everyone_is_a_traitor_controller)
 			if(!is_funmin)
 				return
 			SSblackbox.record_feedback("nested tally", "admin_secrets_fun_used", 1, list("Break All Lights"))
-			message_admins("[key_name_admin(holder)] broke all lights")
+			var/station_only = tgui_alert(usr,"Break which lights?", "And the lord said...", list("All Lights", "Station Z Only", "Cancel"))
+			switch(station_only)
+				if("All Lights")
+					station_only = FALSE
+				if("Station Z Only")
+					station_only = TRUE
+				if("Cancel")
+					return
+				else
+					return
+			message_admins("[key_name_admin(holder)] broke [station_only ? "all station" : "all"] lights")
 			for(var/obj/machinery/light/L in GLOB.machines)
+				if(station_only && !is_station_level(L.z))
+					continue
 				L.break_light_tube()
 				stoplag()
 		if("whiteout")
 			if(!is_funmin)
 				return
 			SSblackbox.record_feedback("nested tally", "admin_secrets_fun_used", 1, list("Fix All Lights"))
-			message_admins("[key_name_admin(holder)] fixed all lights")
+			var/station_only = tgui_alert(usr,"Fix which lights?", "And the lord said...", list("All Lights", "Station Z Only", "Cancel"))
+			switch(station_only)
+				if("All Lights")
+					station_only = FALSE
+				if("Station Z Only")
+					station_only = TRUE
+				if("Cancel")
+					return
+				else
+					return
+			message_admins("[key_name_admin(holder)] fixed [station_only ? "all station" : "all"] lights")
 			for(var/obj/machinery/light/L in GLOB.machines)
+				if(station_only && !is_station_level(L.z))
```

Comment:
feels like a waste of time to have to keep evaluating `station_only` even though it's just a boolean... do you think it would be more worthwhile to do an `if(station_only)`/`else` tree with two different for loops? the only reason why i ask is because it's one of those procs that we don't want to choke out the server with (hence the inclusion of the `stoplag()` proc) and it feels awk to have to keep evaluating the same boolean for hundreds and hundreds of for loops

Reply:
i'm mostly limited on just how to do it not that i'm against it or anything